### ISSUE TITLE
Sample usage

### ISSUE THREAD

**Issue by santhoshmathsca** [PagerDuty/scheduler#21]
Thanks for sharing this scheduler.

 I like to use in our application. I am new to Scala, so if you give me an the Sample main class to run this scheduler it will be very helpful for us.

**Comment by DWvanGeest** [comment]
Hey @sann3, thanks for your interest in this project.

We are hoping to publish our Scaladocs externally, but haven't quite got around to it yet. In the meantime, I've replaced all the private links with links to code/docs available in GitHub (see #22).

Regarding sample usage, you're right that we should have something.  Below is the general idea, however:

``` scala
val scheduler = new SchedulerImpl(
      schedulerSettings,
      typesafeConfig,
      schedulerMetrics,
      cassCluster,
      schedulerKeyspace,
      CassandraTaskExecutorService.factory(workersPerSchedulerPartition, taskRunner)
    )(
      schedulerLogging
    )

scheduler.start()

val schedulerClient = new SchedulerClient(
    kafkaProducer,
    schedulerSettings.kafkaTopic,
    schedulerSettings.schedulingGraceWindow,
    schedulerMetrics
  )(schedulerClientLogging)

val task = Task("some-ordering-id", Instant.now(), "some-uniqueness-key", Map("taskName" -> "my-task")

schedulerClient.scheduleTask(task)
```

Admittedly, you'll have to dig on how to build some of these dependencies like `taskRunner`. I will make a TODO for us to build a more comprehensive example project.

**Comment by sann3** [comment]
Hi @DWvanGeest thanks for the sample code. I saw your speech in the Strange Loop. It was very good speech and very clear, so only I start exploring this project.

Thanks for sharing this great project as open source.

**Comment by kumar-cherry** [comment]
Hi @DWvanGeest any sample application available. thanks

**Comment by kumar-cherry** [comment]
Hi @sann3  you did any sample application.

**Comment by sann3** [comment]
No @kumar-cherry, I didn't created. Just waiting for long time.

**Comment by DWvanGeest** [comment]
Sorry for the late response here, I was on vacation. We do not have a sample application available yet, unfortunately, but it's still on the to-do list.

**Comment by kumar-cherry** [comment]
Any idea at what time we will get.

On 03-Aug-2017 8:10 PM, David van Geest <notifications@github.com> wrote:

Sorry for the late response here, I was on vacation. We do not have a sample application available yet, unfortunately, but it's still on the to-do list.

**Comment by DWvanGeest** [comment]
No, we don't have an expected time to deliver a sample application. As a team we have plenty of other priorities at the moment. You would be welcome to contribute a sample application, which we could review and link to.

**Comment by kumar-cherry** [comment]
OK thanks I will try my best, can you share some more additional details or document how to use library, it will helpful us.

On 03-Aug-2017 9:09 PM, David van Geest <notifications@github.com> wrote:

No, we don't have an expected time to deliver a sample application. As a team we have plenty of other priorities at the moment. You would be welcome to contribute a sample application, which we could review and link to.

**Comment by noorul** [comment]
@DWvanGeest Is it possible to create a how-to document to start using this library? Or a reference app will be even better.
